src/main/java/org/opensearch/geospatial/ip2geo/processor/Ip2GeoProcessor.java

```diff
@@ -175,11 +175,19 @@ private Map<String, Object> filteredGeoData(final Map<String, Object> geoData) {
 
     private void validateDatasourceIsInAvailableState(final String datasourceName) {
         if (ip2GeoCachedDao.has(datasourceName) == false) {
-            throw new IllegalStateException("datasource does not exist");
+            throw new IllegalStateException(String.format(Locale.ROOT, "datasource %s does not exist", datasourceName));
```

Comment:
In general, we don't return customer provided input to the exception message due to security concern because the value is not validated and might cause some security issue.